### PR TITLE
52 download files edit files

### PR DESCRIPTION
**Pull request by camdnnn**
### Overview
Added the download modal, refactoring the file storage component in the process and other front-end components

### What is Implemented
- [x] Download Modal which users can download csv files from using the file select menu

### Minor Improvements
- Refactored front-end folder to add sub-folders for modals
- moved files query to file store component
- Reworked modal state to allow only 1 open modal at a time

### CODE REVIEW

File: API/src/main/java/backend/API/FileUploadController.java [hunk 1]
```diff
@@ -37,6 +37,8 @@
 import org.springframework.web.servlet.HandlerMapping;
 import org.springframework.web.servlet.mvc.method.annotation.MvcUriComponentsBuilder;
 import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.HandlerMapping;
```
Comment (gr812b) [inline API/src/main/java/backend/API/FileUploadController.java:40]: duplicate import from line 37

File: front-end/src/components/App.js [hunk 2]
```diff
@@ -1,32 +1,17 @@
...
-import { UploadModal } from "./uploadModal";
-import { HelpModal } from "./helpModal";
+import { CreateGraphModal } from "./modal/create/CreateGraphModal";
+import { UploadModal } from "./modal/upload/uploadModal";
+import { HelpModal } from "./modal/help/helpModal";
 import React, { useEffect, useState } from 'react';
 import Chart from './Chart';
 import Topbar from './Topbar';
 import $ from 'jquery';
+import { DownloadModal } from './modal/download/downloadModal';
```
Comment (gr812b) [inline front-end/src/components/App.js:9]: No reason to not import after all the other modals (on line 5) to keep it clean

File: front-end/src/components/App.js [hunk 3]
```diff
@@ -70,12 +55,13 @@ const App = () => {
...
   return (
     <div className="App">
-      <Topbar openCreateGraphModal={openCreateGraphModal} openUploadModal={openUploadModal} openHelpModal={openHelpModal}/>
+      <Topbar setModal={setModal}/>
       <header className="App-header">
         <div className="success">{successMessage.message}</div>
-        {showCreateGraphModal ? <CreateGraphModal setShowModal={setShowCreateGraphModal} setChartInformation={setChartInformation} setSuccessMessage={setSuccessMessage}/> : null}
-        {showUploadModal ? <UploadModal setShowUploadModal={setShowUploadModal} setSuccessMessage={setSuccessMessage}/> : null}
-        {showHelpModal ? <HelpModal setShowHelpModal={setShowHelpModal} /> : null}
+        {modal === 'Create' ? <CreateGraphModal setModal={setModal} setChartInformation={setChartInformation} setSuccessMessage={setSuccessMessage}/> : null}
```
Comment (gr812b) [inline front-end/src/components/App.js:61]: I like this, very nice

**Review by gr812b — CHANGES_REQUESTED** [review]
Two tiny changes, I havent tested so I will do that when I get back but make these changes and we're happy and pulling to main!

File: front-end/src/components/App.js (hunk 3 above)
Comment (camdnnn) [reply front-end/src/components/App.js:61]: thank you! (is that good to mark as resolved?)

**Review by gr812b — APPROVED** [review]
Looks great! Merging!

**Pull request merged**